server.py

```diff
@@ -65,215 +57,364 @@ def create_test_room():
     LOG.info(
         request.access_route[0] + ' requested ' + request.url)
     check_new_ip(datetime.now(tz_TO).strftime(
         '%Y-%m-%d %H:%M:%S'), visiting_ips, request.access_route[0])
-    if request.method == 'OPTIONS':
-        return build_preflight_response()
-    elif request.method == 'GET':
-        roomId = request.args.get('roomId')
-        roomFound = False
-        for i in database:
-            if i.id == roomId:
-                roomFound = True
-                i.polling = True
-                return build_actual_response(json.dumps(i, indent=4, cls=RoomEncoder), 200)
-        if roomId is None or roomFound is False:
-            return build_actual_response({"message": "Not Found"}, 404)
+    userId = request.cookies.get('userId')
+    roomId = request.args.get('roomId')
+    try:
+        room = Room.objects.get(roomId=roomId)
+        room.polling = True
+        room.save()
+        room_data = json.loads(room.to_json())
+        return build_response(room_data, 200)
+    except DoesNotExist:
+        return build_response({"message": "Not Found"}, 400)
+    except Exception as e:
+        print(e)
+        return build_response({"message": "Unexpected server error"}, 500)
 
 
 # create a new room object in database.json
 @app.route('/actions/create-room', methods=['POST', 'OPTIONS'])
 def create_room():
     LOG.info(request.access_route[0] + ' requested ' + request.url)
-    global database
-    if request.method == 'OPTIONS':
-        return build_preflight_response()
-    elif request.method == 'POST':
-        database, new_room = write_new_room(
-            database, request.form.get('numMafia'))
-        return build_actual_response({
+    try:
+        numMafia = request.form.get('numMafia')
+        new_room = Room(
+            roomId=generateGameRoomKey(),
+            numMafia=numMafia,
+            night=0,
+            players=[],
+            targets=Target(killTarget='', healTarget='', checkTarget=''),
+            status='pre-game',
+            phase='pre-game',
+            polling=False,
+            roomMaster=uuid.uuid4().hex,
+            votes=[],
+            gameMessages=[GameMessage(
+                primary='Pre-Game', secondary='Waiting for players...')],
+            observerMessages=[ObserverMessage(
+                primary='Pre-Game', secondary='Waiting for players...')]
+        )
+        new_room.save()
+        return build_response({
             "message": "Room created",
-            "roomId": new_room.id
+            "roomId": new_room.roomId
         }, 201, setCookie=True, cookie=new_room.roomMaster)
+    except ValidationError:
+        return build_response({"message": "invalid data"}, 400)
+    except Exception as e:
+        print(e)
+        return build_response({"message": "Unexpected server error"}, 500)
 
 
 # add player object to room
-@app.route('/actions/join-room', methods=['POST', 'OPTIONS'])
+@ app.route('/actions/join-room', methods=['POST', 'OPTIONS'])
```

Comment:
i'm guessing it works with/without a space after the `@`? i see it like this in a few places
```suggestion
@app.route('/actions/join-room', methods=['POST', 'OPTIONS'])
```

Reply:
idk I guess vscode formatting added the space I think